### PR TITLE
Use gravatar for comments

### PR DESCRIPTION
**Pull request by sergous**
https://github.com/dtpstat/dtp-map/issues/81

- Remove avatar default picture
- Remove avatarUrl field from Comments
- Add react-gravatar package

Снимок экрана 2022-01-15 в 9 09 58 PM

### CODE REVIEW

File: src/components/avatar-image.tsx [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import * as React from "react";
+import Gravatar from "react-gravatar";
 import styled from "styled-components";
 
-const defaultAvatarUrl = "/icons/avatar_default.svg";
-
-const StyledImg = styled.img`
+const StyledImg = styled(Gravatar)`
```
Comment (kachkaev) [inline src/components/avatar-image.tsx:5]:
```suggestion
const StyledGravatar = styled(Gravatar)`
```

(`X` → `StyledX` for improved communication)

File: src/components/avatar-image.tsx [hunk 2]
```diff
@@ -14,15 +13,15 @@ const ImageContainer = styled.div`
...
 }
 
 export const AvatarImage: React.VoidFunctionComponent<AvatarImageProps> = ({
-  src,
+  email,
 }) => {
   return (
     <ImageContainer>
-      <StyledImg src={src || defaultAvatarUrl} />
+      <StyledImg email={email || ""} default="wavatar" rating="g" />
```
Comment (kachkaev) [inline src/components/avatar-image.tsx:24]:
```suggestion
      <StyledGravatar email={email || ""} default="mp" rating="g" />
```

Looking at the [docs](https://en.gravatar.com/site/implement/images/), `default=wavatar` may look to funny for our context.

image

`default=mp` looks neutral:

image

File: src/components/user-profile.tsx [hunk 3]
```diff
@@ -15,7 +16,12 @@ export const UserProfile: React.VoidFunctionComponent = () => {
 
   return user ? (
     <div>
-      <img src={user.avatarUrl || ""} alt={user.name || ""} />
+      <Gravatar
+        email={user.email || ""}
+        default="wavatar"
```
Comment (kachkaev) [inline src/components/user-profile.tsx:21]:
```suggestion
        default="mp"
```

**Review by kachkaev — COMMENTED** [review]
## 👍

File: src/components/comment-item.tsx [hunk 4]
```diff
@@ -57,7 +57,7 @@ export const CommentItem: React.VoidFunctionComponent<CommentItemProps> = ({
 }) => {
   return (
     <CommentContainer>
-      <AvatarImage src={comment.author?.avatarUrl} />
+      <AvatarImage email={comment.author?.email} />
       <div>
         <div>
           <CommentAuthor>{comment.author?.name}</CommentAuthor>:{" "}
```
Comment (kachkaev) [inline src/components/comment-item.tsx:63]:
```suggestion
          <CommentAuthor>{comment.author?.name}</CommentAuthor>{" "}
```

Unrelated, optional. Less visual noise → slightly better UI.

**Comment by sergous** [comment]
Recreate the dev branch

**Review by kachkaev — APPROVED** [review]

**Pull request merged**